Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Scene.h

```diff
@@ -281,6 +284,14 @@ namespace AZ
         };
 
         // --- Template functions ---
+
+        template<typename Predicate>
+        FeatureProcessor* Scene::GetFeatureProcessor(const Predicate& predicate) const
+        {
+            auto foundFP = AZStd::find_if(AZStd::begin(m_featureProcessors), AZStd::end(m_featureProcessors), predicate);
```

Comment:
FYI you can use `AZStd::ranges_find_if` to pass in a container instead of 2 iterators
It is slightly shorter.

```suggestion
            auto foundFP = AZStd::ranges::find_if(m_featureProcessors, predicate);
```

Reply:
Alternatively, why not make this into a visitor function which accepts a a callback.

That way lambda could decide to assign the `FeatureProcessor*` to a bound variable.
Also this function wouldn't need to be templated in that instance and can be implemented in Scene.cpp
```
//! Callback function that will be invoked with each non-pointer FeatureProcessor
//! return true to continue visiting or false to halt
using FeatureProcessorVisitCallback = AZStd::function<bool(FeatureProcessor&)>;
// ...

void Scene::VisitFeatureProcessor(FeatureProcessorVisitCallback callback)
{
    auto VisitInterface = [&callback](const FeatureProcessorPtr& featureProcessor
    {
        return featureProcessor != nullptr ? callback(*featureProcessor) : true;
    };

    AZStd::ranges::all_of(m_featureProcessors, VisitInterface);
}
```

Reply:
At first I wasn't inclined to do the visitor pattern since it's just a private function used in a couple places that don't need that functionality, but I suppose I could make it public which may be handy for outside users of Scene.